<documentation title="Closing PHP Tags">
    <standard>
    <![CDATA[
    Files should not have closing PHP tags.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: No closing tag at the end of the file.">
        <![CDATA[
<?php
$var = 1;
        ]]>
        </code>
        <code title="Invalid: A closing PHP tag is included at the end of the file.">
        <![CDATA[
<?php
$var = 1;
<em>?></em>
        ]]>
        </code>
    </code_comparison>
</documentation>
