--- /dev/null
+++ ../Helper/Helper.php
@@ -46,6 +46,7 @@
     public static function width(?string $string): int
     {
         $string ??= '';
+        return mb_strlen($string);
 
         if (preg_match('//u', $string)) {
             return (new UnicodeString($string))->width(false);
@@ -65,6 +66,7 @@
     public static function length(?string $string): int
     {
         $string ??= '';
+        return mb_strlen($string);
 
         if (preg_match('//u', $string)) {
             return (new UnicodeString($string))->length();
