<documentation title="Code Comparison, missing code element">
    <standard>
    <![CDATA[
    This is a standard block.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: There must always be two code elements.">
        <![CDATA[
$valid = true;
        ]]>
        </code>
    </code_comparison>
</documentation>
