<documentation title="Code Comparison Only, Missing Standard Block">
    <code_comparison>
        <code title="Valid: Lorem ipsum dolor sit amet.">
        <![CDATA[
<em>class Code</em> {}
        ]]>
        </code>
        <code title="Invalid: Maecenas non rutrum dolor.">
        <![CDATA[
<em>class Comparison</em> {}
        ]]>
        </code>
    </code_comparison>
</documentation>
